GitCommands/RevisionReader.cs

```diff
@@ -511,34 +466,82 @@ public StringLineReader(string s)
 
                 if (endIndex == -1)
                 {
-                    return ReadToEnd(advance);
+                    return ReadToEnd();
                 }
 
-                if (advance)
-                {
-                    _index = endIndex + 1;
-                }
+                _index = endIndex + 1;
 
                 return pool is not null
                     ? pool.Intern(_s, startIndex, endIndex - startIndex)
                     : _s.Substring(startIndex, endIndex - startIndex);
             }
 
-            public string? ReadToEnd(bool advance = true)
+            public StringLineReader TrimStart()
+            {
+                while (_index < _s.Length && char.IsWhiteSpace(_s[_index]))
+                {
+                    _index++;
+                }
+
+                return this;
+            }
+
+            private void TrimEnd(ref int endIndex)
+            {
+                while (endIndex > _index && char.IsWhiteSpace(_s[endIndex - 1]))
```

Comment:
This is a private method and endIndex is found in _s, so the check will not do anything.
(I made it private instead of a submethod to `ReadToEndOfBody()` to keep it next to TrimStart()